cachi2/core/errors.py

```diff
@@ -126,3 +126,12 @@ class YarnCommandError(Cachi2Error):
         The output of the failing yarn command should provide more details, please check the logs.
         """
     ).strip()
+
+
+class ExternalConfigurationError(Cachi2Error):
```

Comment:
https://github.com/containerbuildsystem/cachi2/pull/413/commits/7d46af459ac4a2f0a458be50f5ec6a522d5cc51c is merging `YarnCommandError` with `GoModError` into a single `SubProcessError`. While I'm not much in favour of that name either, we should leverage that unified exception type for your use case too. So, we either need to merge #413 first or the first 2 patches of that PR would need to be split and proposed separately so that you don't wait indefinitely or you could adopt those patches here in your PR.

Reply:
In this case though, it's not the yarn command itself that is failing, it's that corepack didn't install the correct version of yarn and we don't get the response from yarn that we're looking for. Do we want to make a distinction?

Reply:
[b57270](https://github.com/containerbuildsystem/cachi2/pull/413/commits/b5727071e4e9f878e5071fb84af947bb7be46658) changed `SubProcessError` for `PackageManagerError`. I agree we could use it here for the sake of simplicity.

Reply:
Rebased on that change and switched the exceptions in this PR to PackageManagerError